### ISSUE TITLE
[APF Odisha Phase 2]: Prod Bug fixes - Batch 7: App Performance Issue

### ISSUE THREAD

**Issue by AnjaliBhagabati** [avniproject/apfodishanutrition#435]
- [ ] Changes to be made in UAT First

- [ ] The loading time for the app’s dashboard and home page are longer than expected for Supervisor Poshan Sathis. 
Please find the screen recording shared by one of the users.
User ID of Supervisor who is under the Avni User Group- Poshan Sathi : 9861762465@apfodisha
Screen recording : https://drive.google.com/open?id=1oCLZ3me089zk-wp79ZIe7zERINl32TWv

Note: As per APF team, for QRT Users and Poshan Sathi users (who are not Supervisors) it is taking around 10 secs to load

## Tech Solution
As a part of this card we are moving complex logics from reports cards to the mobile app (using decision)
We have identified the cards which are taking more time than expected are:

- Growth Faltering-1
- SAM children
- Growth Faltering-2
- MAM children
- Total children enrolled in CMAM program
So we have moved the logics for this cards in app itself

1. Growth Faltering-1 & Growth Faltering-2
   Decision Addition: A new decision, "Growth Faltering" has been added in registration.
   Data Handling: We are now pushing updated values from Growth Monitoring encounters to the registration whenever a 
 new encounter occurs. This ensures that the registration decision reflects the most current data.

2. SAM children & MAM children
     Decision Addition: A new decision, "Nutritional Status" has been added in registration.
   Data Handling: We are now pushing updated values from Growth Monitoring and QRT Child encounters to the registration whenever a 
 new encounter occurs. This ensures that the registration decision reflects the most current data.

3. Total children enrolled in CMAM program
    Decision Addition: A new decision, "Is the child enrolled in the CMAM program?" has been added in registration.
   Data Handling: We are now pushing updated values from Growth Monitoring encounters to the registration whenever a 
 new encounter occurs. This ensures that the registration decision reflects the most current data.

**QA Notes:**

1. add data on uat environment and check reg. decision it should updated as per latest encounter

**Comment by sachsk** [comment]
I have reviewed the filters, and there doesn't seem to be any issue with them. It could be a device-related problem.

**Comment by dinesh2096** [comment]
@sachsk @adamsanadi6  closing this card since we didn't make any change and I think testing is also not needed if you think we need to test please fell free to reopen the ticket

**Comment by AnjaliBhagabati** [comment]
@sachsk reopening this card for your review as discussed on today's morning call.

Here are list of other Supervisor with their device details who is facing the performance issue
https://docs.google.com/spreadsheets/d/1hQ4pz36UMBVZY-63dubfV-bSXkgcK14AmIJTfihS67E/edit?gid=0#gid=0

cc: @dinesh2096

**Comment by dinesh2096** [comment]
Test Scenarios : 

1. Complete the growth monitoring and validate that the data entered is reflected in the decision summary.
2. Update the growth monitoring and validate that the data is updated as expected (eg: In the growth monitoring we have the nutrition status as SAM and you have updated to MAM validate the summary is updated to MAM)
3. Edit the growth monitoring and validate that the data is not updated in summary (eg: we have completed the growth monitoring with status SAM and In the next follow-up we have updated the status as MAM if you go back and edit the first encounter the summary page still displays MAM )
4. if we complete the QRT follow-up the value should not get updated.
